modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/FilterToSqlHelper.java

```diff
@@ -930,6 +934,20 @@ private String getPrimaryKeyColumnsAsCommaSeparatedList(
         return sb.toString();
     }
 
+    public Object visit(FilterFunction_arrayAnyMatch filter, Object extraData) {
+        Expression array = getParameter(filter, 0, true);
+        Expression candidate = getParameter(filter, 1, true);
```

Comment:
Do you need to check that the type of values match, will SQL be upset if you check if a boolean is in an array of strings?

Reply:
I am going to add checks for type matching and casting when needed